tests/test_context.py

```diff
@@ -48,3 +49,56 @@ def test_message_handler(device):
     assert context.message_handler is message_handler_test
     with context:
         assert current_context().message_handler is context.message_handler
+
+
+def test_async_wake_interval(device):
+    """Test read-write property async_wake_interval."""
+    with Context(device) as context:
+        context.async_wake_interval = 42
+        assert context.async_wake_interval == 42
+
+
+# THIS DOES NOT WORK, PLEASE HELP!
+
+# def test_is_supported(device):
+#     with Context(device) as context:
+#         assert context == current_context()
```

Comment:
I don't think we need to assert it here any more.

Reply:
I second this, please remove these unless the test is specifically to assert it.

Reply:
To quote `is_supported` docstring:
```
        """Return if the channel configuration and sample type
        are supported by the context.

        This method require the context to be current.

        See Also
        --------
        sample_types : Set of sample types
        channel_configs : Set of channel configurations
        """
```

Reply:
`with Context(device) as context` already make `context` current.  You can test for the other case outside of the `with` block.